### PR TITLE
Update method for computing piano rolls

### PR DESCRIPTION
**Pull request by neosatrapahereje**
This pull request solves issue #18. 

New changes:

* The method `notearray_to_pianoroll` was renamed to `compute_pianoroll`
* The method accepts now structured arrays (with the new time units), `Part`, `PartGroup`, `PerformedPart` objects, as well as lists of `Part` objects.
* All parts, channels are flattened into a single piano roll. 
* There is an option to not consider the notes in the drum track (i.e., channel 9 in a MIDI file) from the piano roll

### CODE REVIEW

File: partitura/performance.py [hunk 1]
```diff
@@ -137,18 +141,31 @@ def from_note_array(cls, note_array,
...
+
+        if 'channel' not in note_array.dtype.names:
+            channel = np.ones(len(note_array), dtype=int)
+        else:
+            channel = note_array['channel']
+
         notes = []
-        for nid, note in zip(n_ids, note_array):
+        for nid, note, track, channel in zip(n_ids, note_array,
+                                             track, channel):
```
Comment (mgrachten) [inline partitura/performance.py:161]: it's a bit confusing to reuse the variable names `track`/`channel` in the for loop.

**Review by mgrachten — APPROVED** [review]
Apart from some minor issues (see comments), looks good.

**Comment by mgrachten** [comment]
Could you run `make doctest` from the `docs` folder to make sure that the real outputs of coincide with the outputs written in the rst file? I get 5 failures.

File: partitura/musicanalysis/voice_separation.py [hunk 2]
```diff
@@ -23,58 +28,51 @@ def rename_voices(voices):
...
 def argmax_pitch(idx, pitches):
     return idx[np.argmax(pitches[idx])]
 
 
-def estimate_voices(notearray, monophonic_voices=False):
-    """Voice estimation using the voice separation algorithm proposed 
-    in [1]_.
+def estimate_voices(note_info, monophonic_voices=False):
+    """Voice estimation using the voice separation algorithm
+       proposed in [1]_.
```
Comment (mgrachten) [inline partitura/musicanalysis/voice_separation.py:64]: Could you use the number 6 for the reference? That way there are no collisions with the references when the docstrings of all the musicanalysis functions are aggregated into a single html file.

File: partitura/musicanalysis/voice_separation.py [hunk 3]
```diff
@@ -89,8 +87,8 @@ def estimate_voices(notearray, monophonic_voices=False):
     References
     ----------
     .. [1] Chew, E. and Wu, Xiaodan (2004) "Separating Voices in
```
Comment (mgrachten) [inline partitura/musicanalysis/voice_separation.py:89]: Should also be changed to 6

File: partitura/musicanalysis/pitch_spelling.py [hunk 4]
```diff
@@ -9,53 +9,64 @@
...
+ChromamorpheticPitch = namedtuple('ChromamorpheticPitch',
+                                  'chromatic_pitch morphetic_pitch')
 
 STEPS = np.array(['A', 'B', 'C', 'D', 'E', 'F', 'G'])
 UND_CHROMA = np.array([0, 2, 3, 5, 7, 8, 10], dtype=np.int)
 ALTER = np.array(['n', '#', 'b'])
 
 
-def estimate_spelling(note_array, method='ps13s1', **kwargs):
+def estimate_spelling(note_info, method='ps13s1', *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (neosatrapahereje) [reply partitura/musicanalysis/pitch_spelling.py]: Fixed with commit 8008d74

File: docs/usage.rst [hunk 5]
```diff
@@ -461,10 +462,45 @@ The number of sharps/flats and the mode can be inferred from the key name using
...
 
+Pitch Spelling
+--------------
+
+Pitch spelling estimation is performed by the function
+:func:`~partitura.musicanalysis.estimate_spelling`. The function returns a structured array with pitch spelling information (i.e., with fields `step`, `alter` and `octave`) for each note in the input `note_array`. If the input to this method is an instance of :class:`~partitura.score.Part`,  :class:`~partitura.score.PartGroup`, or :class:`~partitura.performance.PerformedPart`, a list of :class:`~partitura.score.Part`, each row of the output corresponds to order of the notes in the `note_array` that would be generated by using the helper method :func:`~partitura.utils.ensure_notearray`.
+
+>>> pitch_spelling = partitura.musicanalysis.estimate_spelling(part.note_array)
+>>> print(pitch_spelling)
+[('A', 0, 4) ('C', 0, 5) ('E', 0, 5)]
```
[the remark this answers is not in the document]
Comment (neosatrapahereje) [reply docs/usage.rst]: Fixed all issues with doctests in commit ae56f35

File: partitura/musicanalysis/key_identification.py [hunk 6]
```diff
@@ -94,37 +100,45 @@ def build_key_profile_matrix(key_prof_maj, key_prof_min):
...
     ----------
     .. [2] Krumhansl, Carol L. (1990) "Cognitive foundations of musical pitch",
            Oxford University Press, New York.
     .. [3] Temperley, D. (1999) "What's key for key? The Krumhansl-Schmuckler
-           key-finding algorithm reconsidered". Music Perception. 17(1), 
+           key-finding algorithm reconsidered". Music Perception. 17(1),
            pp. 65--100.
 
     """
+    from partitura.utils.music import ensure_notearray
```
[the remark this answers is not in the document]
Comment (neosatrapahereje) [reply partitura/musicanalysis/key_identification.py]: Fixed in commit 3fd2eb6.

File: partitura/musicanalysis/tonal_tension.py [hunk 7]
```diff
@@ -233,7 +241,8 @@ class CloudDiameter(TonalTension):
     """
     Compute cloud diameter
     """
-    def compute_tension(self, cloud, scale_factor=SCALE_FACTOR, *args, **kwargs):
+    def compute_tension(self, cloud, scale_factor=SCALE_FACTOR,
+                        *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (neosatrapahereje) [reply partitura/musicanalysis/tonal_tension.py]: Fixed in 3cdc757.

File: partitura/musicanalysis/voice_separation.py (hunk 2 above)
Comment (neosatrapahereje) [reply partitura/musicanalysis/voice_separation.py]: Fixed in b9a7917.

File: partitura/musicanalysis/voice_separation.py (hunk 3 above)
Comment (neosatrapahereje) [reply partitura/musicanalysis/voice_separation.py]: Fixed in b9a7917.

File: partitura/performance.py (hunk 1 above)
Comment (neosatrapahereje) [reply partitura/performance.py]: Fixed in a4fcd88.

**Pull request merged**